setup.py

```diff
@@ -185,6 +162,7 @@ def recursive_data_files(directory, install_directory):
         + recursive_data_files("keyrings", "share/snapcraft")
         + recursive_data_files("extensions", "share/snapcraft")
     ),
+    python_requires=">=3.10",
```

Comment:
Reply to your PR description:

Yes, you made the right choice since we build snapcraft on 3.10.

Reply:
We probably should have done this when we changed the linters over 😆